File: bloaters/long-method/introduce-parameter-object/PHP/problem.php
<?php

namespace PHP;

use DateInterval;
use DateTime;

class Problem
{
    public $invoicedInStartDate;
    public $invoicedInEndDate;
    public $receivedInStartDate;
    public $receivedInEndDate;

    function __construct()
    {
    }

    public function printLogs(): void
    {
        $this->invoicedIn();
        $this->receivedIn();
    }

    private function invoicedIn(): void
    {
        echo "The invoice duration is: {$this->totalDuration($this->invoicedInStartDate,$this->invoicedInEndDate)->format('%R%a day(s)')}\n";
    }

    private function receivedIn(): void
    {
        echo "The receipt duration is: {$this->totalDuration($this->receivedInStartDate,$this->receivedInEndDate)->format('%R%a day(s)')}\n";
    }

    private function totalDuration(DateTime $startDate, DateTime $endDate): DateInterval
    {
        return $startDate->diff($endDate);
    }
}

File: bloaters/long-method/introduce-parameter-object/C#/Problem.cs
using System;

namespace CSharp
{
    public class Problem
    {
        public DateTime InvoicedInStartDate { get; set; }
        public DateTime InvoicedInEndDate { get; set; }
        public DateTime ReceivedInStartDate { get; set; }
        public DateTime ReceivedInEndDate { get; set; }

        public Problem()
        {
        }

        public void PrintLogs()
        {
            InvoicedIn();
            ReceivedIn();
        }

        private void InvoicedIn()
        {
            Console.WriteLine($"The invoice duration is: {TotalDuration(InvoicedInStartDate, InvoicedInEndDate).TotalDays} day(s)");
        }

        private void ReceivedIn()
        {
            Console.WriteLine($"The receipt duration is: {TotalDuration(ReceivedInStartDate, ReceivedInEndDate).TotalDays} day(s)");
        }

        private TimeSpan TotalDuration(DateTime start, DateTime end)
        {
            return end - start;
        }
    }
}
